src/adapter/src/coord/sequencer/inner/create_view.rs

```diff
@@ -60,17 +58,23 @@ impl Coordinator {
                 return_if_err!(validity.check(self.catalog()), ctx);
             }
 
-            (ctx, stage) = match stage {
+            (ctx, stage, span) = match stage {
                 Validate(stage) => {
+                    let span = span.entered();
```

Comment:
Manually entering and exiting spans is tricky business, because someone could insert a `.await` call in-between which would [mess up the trace](https://docs.rs/tracing/latest/tracing/struct.Span.html#in-asynchronous-code). The `return_if_err!` macro throws a wrench in this though because it explicitly `return`s from the whole method. What do you think about the following refactor?

```
let result = span.in_scope(|| {
  self.create_view_validate(ctx.session(), stage)
});
let next = return_if_err!(result);

(ctx, CreateViewStage::Optimize(next), span)
```